File: app/src/main/java/com/annapol04/munchkin/db/ActionTypeConverter.java
package com.annapol04.munchkin.db;

import android.arch.persistence.room.TypeConverter;

import com.annapol04.munchkin.engine.Action;

public class ActionTypeConverter {

    @TypeConverter
    public static int toInt(Action action) {
        return action.getId();
    }

    @TypeConverter
    public static Action toAction(int id) {
        return Action.Companion.fromId(id);
    }
}

File: app/src/main/java/com/annapol04/munchkin/db/ActionTypeConverter.kt
package com.annapol04.munchkin.db

import android.arch.persistence.room.TypeConverter

import com.annapol04.munchkin.engine.Action

class ActionTypeConverter {

    @TypeConverter
    fun toInt(action: Action): Int {
        return action.id
    }

    @TypeConverter
    fun toAction(id: Int): Action {
        return Action.fromId(id)
    }
}
